lib/viewport.dart

```diff
@@ -1,85 +1,45 @@
-import 'dart:ui';
-
+import 'package:flame/game.dart';
 import 'package:forge2d/forge2d.dart';
 
-import 'body_component.dart';
-
-class Viewport extends ViewportTransform {
-  Vector2 size;
-
-  @override
-  double scale;
+mixin Forge2DViewport on Viewport {
+  late ViewportTransform viewportTransform;
 
-  Viewport(this.size, this.scale) : super(size / 2, size / 2, scale);
+  double get scale => viewportTransform.scale;
+}
 
-  double worldAlignBottom(double height) => -(size.x / 2 / scale) + height;
+class Forge2DDefaultViewport extends DefaultViewport with Forge2DViewport {
+  Forge2DDefaultViewport(double scale) {
+    viewportTransform = ViewportTransform(
+      Vector2.zero(),
+      Vector2.zero(),
+      scale,
+    );
+  }
 
   /// Resizes the current viewport.
+  @override
   void resize(Vector2 size) {
```

Comment:
how does this work on web when I zoom in and out. It seems the body would move around because it does not take translation into account

Reply:
How do you mean that the body would move around, screen-wise or world-wise?

Reply:
Or is this what you refer to?
https://github.com/flame-engine/flame_forge2d/pull/32#issuecomment-802380111

Reply:
When player moves center also moves. On resize should take that move into consideration but you are resetting it here

Reply:
If you are following the player that shouldn't be a problem, it will be updated within the same tick?

Reply:
Ah, you are correct, I'll update this!